### PR TITLE
Update `SyncSitesProvider` types

### PR DESCRIPTION
**Pull request by fredrikekelund**
## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/10088

## Proposed Changes

The type definitions in `SyncSitesProvider` and the related "sub-hooks" are unnecessarily complex. They cross-reference in ways that make them difficult to reason about.

This PR adds explicit return type definitions for `useSyncPull`, `useSyncPush`, and `useSiteSyncManagement`. I've also cleaned up `SyncSitesContextType` and excluded state and functions that weren't being used anywhere in the codebase.

## Testing Instructions

Code review and smoke testing the sync flow should suffice

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/components/sync-connected-sites.tsx [hunk 1]
```diff
@@ -188,7 +182,10 @@ const SyncConnectedSitesSection = ( {
 					const hasPullFinished = sitePullState && isKeyFinished( sitePullState.status.key );
 
 					const pushState = getPushState( selectedSite.id, connectedSite.id );
-					const isPushError = pushState.isError;
+					const isPushing = pushState && isKeyPushing( pushState.status.key );
+					const isPushError = pushState && isKeyFailed( pushState.status.key );
+					const hasPushFinished = pushState && isKeyFinished( pushState.status.key );
+
```
Comment (fredrikekelund) [inline src/components/sync-connected-sites.tsx:212]: This change ensures that `getPullState` and `getPushState` can have the same return type. Previously, `getPushState` would add a few attributes to its return value to include these details. I don't see any good reason for treating `getPullState` and `getPushState` differently, though.

File: src/components/sync-connected-sites.tsx [hunk 2]
```diff
@@ -147,7 +141,7 @@ const SyncConnectedSitesSection = ( {
 					variant="link"
 					className="!ml-auto !text-a8c-gray-70 hover:!text-a8c-red-50 "
 					onClick={ handleDisconnectSite }
-					disabled={ isPulling || isPushing }
+					disabled={ isSiteIdPulling( selectedSite.id ) || isSiteIdPushing( selectedSite.id ) }
```
Comment (fredrikekelund) [inline src/components/sync-connected-sites.tsx]: I believe this should be the equivalent of the previous `isPulling` and `isPushing` definitions.

Comment (katinthehatsite) [reply src/components/sync-connected-sites.tsx]: I think I spotted a small regression with this one - the disconnect button now is disabled for all connected sites on a particular selected site. The expected behaviour is that it should be disabled only for the `Disconnect` button on a site that is syncing. I think this screenshot illustrates the issue:

Screenshot 2024-12-12 at 9 11 03 AM

**Comment by katinthehatsite** [comment]
I looked through changes and tested the flow. It seemed to work as expected besides one small regression I noticed. However, I am not sure if the regression is related to the changes in this PR or to the differences with trunk. Perhaps we could resolve the conflicts with trunk and give it another review? 🙇

File: src/components/sync-connected-sites.tsx (hunk 2 above)
Comment (fredrikekelund) [reply src/components/sync-connected-sites.tsx]: Thanks, great catch 👍 Fixed now

**Comment by fredrikekelund** [comment]
Conflicts fixed and your comment addressed, @katinthehatsite. This should be good for a second look now

File: src/components/content-tab-import-export.tsx [hunk 3]
```diff
@@ -305,18 +304,11 @@ const ImportSite = ( {
...
-		const sitePullState = getPullState( selectedSite.id, site.id );
-		return sitePullState && isKeyPulling( sitePullState.status.key );
-	} );
-	const isPushing = connectedSites.some( ( site ) => {
-		const sitePushState = getPushState( selectedSite.id, site.id );
-		return sitePushState?.isInProgress;
-	} );
+	const isPulling = connectedSites.some( ( site ) => isSiteIdPulling( selectedSite.id, site.id ) );
+	const isPushing = connectedSites.some( ( site ) => isSiteIdPushing( selectedSite.id, site.id ) );
 	const isThisSiteSyncing = isPulling || isPushing;
```
Comment (katinthehatsite) [inline src/components/content-tab-import-export.tsx:312]: I see that we are using this logic in a couple of different files. I am wondering if we might want to create a hook to be able to reuse this check. This is not directly related to this PR though as this was already present before. We can create a separate issue to consider this.

**Review by katinthehatsite — APPROVED** [review]
The changes look good to me. I did not spot any other regressions after the conflicts with trunk were resolved 👍 

I had a small suggestion but I do not think it is directly related to this PR.
